### PR TITLE
SNOW-705498: Improve the robustness of ocsp response caching to handle error in cases of serialization and deserialization

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-705498

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [x] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

  1. white list of attributes to pickle
  2. wrap serialization/deserialization to catch all erros
  3. move self._get_item_nonlocking out of __init__ definition to a separate method to avoid being collected

### CODE REVIEW

File: src/snowflake/connector/cache.py [hunk 1]
```diff
@@ -595,8 +614,9 @@ def clear(self) -> None:
 
     def __getstate__(self) -> dict:
         state = self.__dict__.copy()
-        del state["_lock"]
-        del state["_file_lock"]
+        for k in list(state.keys()):
+            if k not in SFDictFileCache._ATTRIBUTES_TO_PICKLE:
+                del state[k]
```
Comment (sfc-gh-sfan) [inline src/snowflake/connector/cache.py:619]: I wonder if we could just do: `state = {k: v for k, v in state.items() if k in SFDictFileCache._ATTRIBUTES_TO_PICKLE}`

File: src/snowflake/connector/version.py [hunk 2]
```diff
@@ -1,3 +1,3 @@
 # Update this for the versions
 # Don't change the forth version number from None
-VERSION = (2, 9, 0, None)
+VERSION = (2, 9, 2, None)
```
Comment (sfc-gh-sfan) [inline src/snowflake/connector/version.py:3]: Curious: Where is 2.9.1 😂

Comment (sfc-gh-aling) [reply src/snowflake/connector/version.py:3]: this PR is expected to be merged after release of 2.9.1

File: src/snowflake/connector/cache.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/connector/cache.py]: thanks for the suggestion, updated implementation

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/connector/cache.py [hunk 3]
```diff
@@ -339,6 +346,15 @@ class SFDictFileCache(SFDictCache):
...
     MAX_RAND_INT = 9
+    _ATTRIBUTES_TO_PICKLE = [
+        "_entry_lifetime",
+        "_cache",
+        "telemetry",
+        "file_path",
+        "file_timeout",
+        "_file_lock_path",
+        "last_loaded",
+    ]
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/cache.py:357]: Immutability
```suggestion
    _ATTRIBUTES_TO_PICKLE = (
        "_entry_lifetime",
        "_cache",
        "telemetry",
        "file_path",
        "file_timeout",
        "_file_lock_path",
        "last_loaded",
    )
```

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

**Review by sfc-gh-aalam — APPROVED** [review]

**Pull request merged**